org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLFormatterTest.java

```diff
@@ -386,14 +386,87 @@ public void testDTDFormatting() throws BadLocationException {
 		String expected = 
 			"<!DOCTYPE web-app PUBLIC" + lineSeparator() +
 			"\"-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN\"" + lineSeparator() +
-			"\"http://java.sun.com/dtd/web-app_2_3.dtd\" >" + lineSeparator() + 
+			"\"http://java.sun.com/dtd/web-app_2_3.dtd\" >" + lineSeparator() +
 			"<web-app>" + lineSeparator() +
 			"  <display-name>Servlet 2.3 Web Application</display-name>" + lineSeparator() +
 			"</web-app>";
 		XMLFormattingOptions formattingOptions = createDefaultFormattingOptions();
 		format(content, expected, formattingOptions);
 	}
 
+	@Test
+	public void testDTDFormattingInternal() throws BadLocationException {
+		String content = 
+			"<?xml version=\"1.0\"?>" + lineSeparator() +

```

Comment:
Please don't use lineSeparator() because your tests depends on OS. Please use \r or \ r\n.

If you use Eclipse to develop Java, you can copy a content of XML file and go to the Java Editor code. Type just "" and copy/paste the pasted XML content inside the "". Java Editor JDT will manage \r

Reply:
I though \r\n was specific to Windows, do you know why it works for me on linux?

Reply:
Sorry with my bad explanation. 

> I though \r\n was specific to Windows, 

Yes it is.

I meant that we have tests which is not linked to an OS. So in your Linux case, you will havelineSparatator= \r and in my OS Windows case, I will have lineSparatator=\r\n So the offset will be different.

I agree with you, your test doesn't use offset so we will hav eno problem, but it should be more clean to have tests which hard coded the line separator.

Hope you will understand my explanation.